### PR TITLE
[WFCORE-5979] Move com.google.guava:guava and com.google.guava:failureaccess to testbom

### PR DESCRIPTION
**Pull request by sudeshnas93**
Issue: https://issues.redhat.com/browse/WFCORE-5979

### CODE REVIEW

File: elytron/pom.xml
Comment (yersan) [inline elytron/pom.xml:4]: FYI, we are not required to update the year of the license header files. Just if you add a new file, use the expected year but we don't need to update the header when we are modifying the existing ones.

File: testsuite/patching/pom.xml
Comment (yersan) [inline testsuite/patching/pom.xml:60]: Could you also add `<scope>test</scope>`?

File: testsuite/patching/pom.xml
Comment (yersan) [inline testsuite/patching/pom.xml:71]: I think this is unnecessary, you have explicitly added before the `com.google.guava:guava` dependency, so the transitive one that could come from `org.apache.maven:maven-resolver-provider` never will be used, can you confirm from your experiments?

File: testbom/pom.xml
Comment (yersan) [inline testbom/pom.xml:63]: Could you also add `<scope>test</scope>`?

File: testbom/pom.xml
Comment (yersan) [inline testbom/pom.xml:95]: Could you also add `<scope>test</scope>`?

**Review by yersan — CHANGES_REQUESTED** [review]
Integration tests with WildFly full are failing with the following message:

`java.lang.NoClassDefFoundError: com/google/common/util/concurrent/internal/InternalFutureFailureAccess`

That means there are modules in WildFly that are using the `com.google.guava: failureaccess` artifact that comes from wildfly-core instead of the one that is using win WildFly Full.

Before being able to merge this PR, we need a Jira to decouple the use of the WildFly Core  `com.google.guava: failureaccess` in favor of the one declared in WildFly main pom.

Once this change is in full, we can proceed with this.

File: elytron/pom.xml
Comment (sudeshnas93) [reply elytron/pom.xml:4]: Okay. Thank you for the information. I will revert it to the original ones.

Comment (iweiss) [reply elytron/pom.xml:4]: I think in the past I understood that we needed to update the copyright date for any existing files we modify. Isn't it?

Comment (yersan) [reply elytron/pom.xml:4]: Hi @iweiss , you can refer to this Zulip conversation where the copyright headers updating need was discussed: https://wildfly.zulipchat.com/#narrow/stream/174184-wildfly-developers/topic/Copyright.20Headers

**Comment by iweiss** [comment]
@sudeshnas93 you need to rebase and then push your branch

File: testbom/pom.xml
Comment (sudeshnas93) [reply testbom/pom.xml:96]: Added

File: testbom/pom.xml
Comment (sudeshnas93) [reply testbom/pom.xml:63]: Added.

File: testsuite/patching/pom.xml
Comment (sudeshnas93) [reply testsuite/patching/pom.xml]: faced maven error while building previously; removed them now, building successfully in local now. So, added the changes.

File: testsuite/patching/pom.xml
Comment (sudeshnas93) [reply testsuite/patching/pom.xml:60]: Added

**Comment by sudeshnas93** [comment]
> Integration tests with WildFly full are failing with the following message:
> 
> `java.lang.NoClassDefFoundError: com/google/common/util/concurrent/internal/InternalFutureFailureAccess`
> 
> That means there are modules in WildFly that are using the `com.google.guava: failureaccess` artifact that comes from wildfly-core instead of the one that is using win WildFly Full.
> 
> Before being able to merge this PR, we need a Jira to decouple the use of the WildFly Core `com.google.guava: failureaccess` in favor of the one declared in WildFly main pom.
> 
> Once this change is in full, we can proceed with this.

I have raised an issue in Wildfly related to this error. I will try to send the PR asap.
The issue link: https://issues.redhat.com/browse/WFLY-16952

**Comment by sudeshnas93** [comment]
[Linking the associated PR](https://github.com/wildfly/wildfly/pull/16088) which needs to be merged before merging #5190

**Comment by bstansberry** [comment]
https://github.com/wildfly/wildfly/pull/16088 is merged

**Comment by yersan** [comment]
Hi @sudeshnas93 notice this is going to conflict due to https://github.com/wildfly/wildfly-core/pull/5237

I have also created https://github.com/wildfly/wildfly/pull/16161, once merged I believe the errors thrown by this PR will be fixed. Could you keep an eye on [WFLY-17128](https://issues.redhat.com/browse/WFLY-17128) and once merged rebase this to check how it goes?

**Comment by sudeshnas93** [comment]
Hello @yersan, I have rebased the PR and resolved the conflicts raised due to the version update of fasterxml.jackson . Also, https://github.com/wildfly/wildfly/pull/16161 has been merged, waiting for CI results.

File: testbom/pom.xml
Comment (yersan) [inline testbom/pom.xml:43]: This is expected to be 31.1-jre, introduced by https://github.com/wildfly/wildfly-core/pull/5237

In addition, you have to remove from the main pom.xml the properties of the artifacts you are moving to test bom:
```
<version.com.google.guava>31.1-jre</version.com.google.guava>
<version.com.google.guava.failureaccess>1.0.1</version.com.google.guava.failureaccess>
```

**Review by yersan — CHANGES_REQUESTED** [review]

File: testbom/pom.xml
Comment (sudeshnas93) [reply testbom/pom.xml]: Made the changes. Thank you. Please review.

**Review by yersan — APPROVED** [review]

**Pull request merged**
